stl/inc/xstring

```diff
@@ -4038,12 +4021,15 @@ public:
         return _Count;
     }
 
-    void _Swap_bx_large_with_small(_Scary_val& _Starts_large, _Scary_val& _Starts_small) {
+    _CONSTEXPR20_CONTAINER void _Swap_bx_large_with_small(_Scary_val& _Starts_large, _Scary_val& _Starts_small) {
         // exchange a string in large mode with one in small mode
-        // (not _CONSTEXPR20_CONTAINER; SSO should be disabled in a constexpr context)
-
         const pointer _Ptr = _Starts_large._Bx._Ptr;
         _Destroy_in_place(_Starts_large._Bx._Ptr);
+#ifdef __cpp_lib_constexpr_string
+        if (_STD is_constant_evaluated()) {
+            _Starts_large._Bx._Buf = {};
+        }
+#endif // __cpp_lib_constexpr_string

```

Comment:
Is there a guarantee that CharType is trivial? If so we should be able to unconditionally create the array without any runtime cost.

Reply:
[strings.general]/1:
> This Clause describes components for manipulating sequences of any non-array trivial standard-layout type. Such types are called _char-like_ _types_, and objects of char-like types are called _char-like_ _objects_ or simply characters.